### ISSUE TITLE
Remove Station Size Limitation

### ISSUE THREAD

**Issue by glenjimen** [OpenLoco/OpenLoco#358]
OS:  Windows 7 & XP
Version:  All versions, Vanilla and OpenLoco
Edit : Forgot how XML comments work, some text was not visible.

  Part of the design of the original (vanilla) Locomotion was that you could only build stations 16 tiles long. There is a trick to getting around this, it involves building a station with most tiles lumped to one side, this forces the station name label to become offset to one side, allowing the player to build more tiles on that side.  It appears you can only build station tiles 8 spaces away from the centered station name label.
  My request is to remove the limitation of 16 station tiles in any direction.  The hard limit of 80 tiles would remain for now, as changing that would probably mess up the save file format.
  Many players like to run long trains and have had to build the longer stations in the frustrating round-about manner. 

**Steps to reproduce:**
1.  Build a station that is more than 16 tiles long.
2.  Try to build longer in either direction and you receive the error "Station too Spread Out"
Note (not part of this issue) : If you build a station more than 80 tiles you get the error "Station too Large"

**Is the issue reproducible in Locomotion (vanilla)?**

Video of player making longer station in vanilla Locomotion:
https://www.youtube.com/watch?v=SzJAAwv_z3g

**Comment by AaronVanGeffen** [comment]
We've yet to implement most of the (station) construction functions, but this is an interesting nuance to #126 for sure. If we can easily patch the relevant checks (that is, not [this many](https://github.com/OpenLoco/OpenLoco/issues/126#issuecomment-466663085)), I'd be happy for us to include such a patch.

**Comment by AaronVanGeffen** [comment]
String 1183 is the "Station too spread out!" error. The following functions refer to it:

```
Address	Function	Instruction
.text:00493EF4	sub_493AA7	mov     game_command_error_reason_stringid, 1183
.text:004930C7	sub_492F3D	mov     game_command_error_reason_stringid, 1183
.text:0048D28E	sub_48C708	mov     game_command_error_reason_stringid, 1183
.text:0048C3E4	sub_48BB20	mov     game_command_error_reason_stringid, 1183
```

Each of these refers to one particular medium of transport. I think we should be able to make do with patching `sub_48BB20`. However, when I just patch out the jump to writing the error code, the game just happily makes a _new_ station instead of extending the initial one…

**Comment by glenjimen** [comment]
I thought this would be a simple issue to patch.  If this gets too complected, it's probably better to wait until the station construction functions are implemented.

The length of station you can build seems to be constrained by the distance each tile is from the station's name label.  This distance is only checked when each new tile is built, and is ignored afterwards.  At least, that's what I've experienced in-game, and what is seen in the attached video (in the first post).
Is there a variable that controls this distance?  Can it simply be increased?

**Comment by AaronVanGeffen** [comment]
> Is there a variable that controls this distance? Can it simply be increased?

Not yet. Until the construction functions are implemented as C++ routines, we have to patch the assembly (machine code). You see how this is a little more complicated than the former. :-)

**Comment by AaronVanGeffen** [comment]
While going over old issues, I figured I'd take another stab at this. I was not successful, but I figured I'd document what I did, anyway.

The function dealing with track station placement is 0x0048BB20. The 'too spread out' error appears at 0x0048C3E4, which is jumped into only from 0x0048BDAB. I patched out the checks by writing a no-op sled from 0x0048BDA4 to 0x0048BDB0.

This successfully prevents the error from showing up, but instead a new station is created. I don't know enough about this game action to stop this from happening. Looking back at what I said 1.5 years ago, it may be about time we get to rewriting the game action in question in C++ :sweat_smile:

**Comment by DriverKaldie** [comment]
And what about the code from the long station patch that I have uploaded with permission from the original creator of that mod on the tt-forums ?

https://www.tt-forums.net/viewtopic.php?f=38&t=50627

**Comment by MVTaxi** [comment]
> While going over old issues, I figured I'd take another stab at this. I was not successful, but I figured I'd document what I did, anyway.
> 
> The function dealing with track station placement is 0x0048BB20. The 'too spread out' error appears at 0x0048C3E4, which is jumped into only from 0x0048BDAB. I patched out the checks by writing a no-op sled from 0x0048BDA4 to 0x0048BDB0.
> 
> This successfully prevents the error from showing up, but instead a new station is created. I don't know enough about this game action to stop this from happening. Looking back at what I said 1.5 years ago, it may be about time we get to rewriting the game action in question in C++ 😅

I found the function which makes "Station too spread out" error. Located in _0x0049184A_(Loco.exe+9184A, Original game, steam version). By replacing this line to `ret` (using Cheat Engine), game does not limit size of a station and player can build up to 80 tiles length station in single line.


![Screenshot](https://user-images.githubusercontent.com/55020165/222967265-ee5cd61e-f35b-427a-8062-c5d52d943bba.png)

**Comment by LeeSpork** [comment]
> My request is to remove the limitation of 16 station tiles in any direction. The hard limit of 80 tiles would remain for now, as changing that would probably mess up the save file format.

I'm confused why this issue is still open. It, exactly as it was described in the initial post, has already been resolved/implemented with the "Disable station size limitations" checkbox that can now be found in the Miscellaneous options.

Although the changelog entry under v24.03 attributes this to #126 for some reason (#126 appears to also be asking for the station tile limit to be increased from 80, which is explicitly stated to be out of scope for this issue).

**Comment by glenjimen** [comment]
Yes, you are correct.  This has now been implemented.  I've marked it closed.
